data_adapter_oemof/calculations.py

```diff
@@ -34,3 +42,283 @@ def get_name(*args, counter=None):
+
+        max = [
+            "potential_annual_max",
+            "capacity_p_max",
+            "capacity_e_max",
+            "capacity_w_max",
+            "availability_timeseries_max",
+            "capacity_tra_connection_max",
+            "flow_share_max_",
+            "sto_cycles_max",
+            "sto_max_timeseries",
+            "capacity_p_abs_new_max",
+            "capacity_e_abs_new_max",
+            "capacity_w_abs_new_max",
+        ]
+
+        for column in group_df.columns:
+            if column in ["method", "source", "comment", "bandwidth_type"]:
+                continue
+
+            """
+            Following is a check whether nans can be filled.
+
+            Commented check for columns that are faulty and need to be changed
+            Commented Error for incomplete columns as we dont know where it may cause errors yet
+
+            """
+            if column in max:
+                group_df[column] = group_df[column].fillna(max_value)
+            elif column in min:
+                group_df[column] = group_df[column].fillna(min_value)
+
+        return group_df
+
+    def find_and_replace_irrelevant_data(group_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Finds and replaces irrelevant Data.
+
+        Searches for where investment is allowed
+            - If no Investment is allowed and there is no period above where investment was allowed
+                -> Data is replaced by pandas ffil/bfill
+            - If no Investment is allowed and there have been periods with investment before
+                -> Data is replaced by pandas ffil/bfill
+
+        Searches for decomissioned Processes
+            - If capacity of a process is 0 other nan data is replaced by mean.
+
+        Parameters
+        ----------
+        group_df
+
+        Returns
+        -------
+
+        """
+
+        capacity_columns = [
+            "capacity_p_inst_0",
+            "capacity_e_inst_0",
+            "capacity_w_inst_0",
+            "capacity_tra_inst_0",
+        ]
+
+        invest_zero = [
+            "capacity_p_abs_new_max",
+            "capacity_e_abs_new_max",
+            "capacity_w_abs_new_max",
+            "capacity_p_max",
+            "capacity_e_max",
+            "capacity_w_max",
+        ]
+        # no investment in decommissioning processes
+        # no capacity can be set on investment objects
+        if not any([x in capacity_columns for x in group_df.columns]):
+            for invest_column in invest_zero:
+                if invest_column in group_df.columns:
+                    non_investment_indices = group_df[invest_column] == 0
+                    # In Case first values are missing bfil is used in case last values missed
+                    # ffil is used. In most cases boundary values are missing
+                    # therefore interpolation is no viable option.
+                    group_df.loc[non_investment_indices] = group_df.bfill().loc[
```

Comment:
you should `loc` first and `fill` afterwards. Otherwise you are filling each column and select needed column afterwards, which could be a big overhead!

Reply:
Thank you:)